controllers/mentor.go

```diff
@@ -106,3 +134,104 @@ func FetchAllMentors(w http.ResponseWriter, r *http.Request) {
 
 	utils.RespondWithJson(r, w, mentors)
 }
+
+// /mentor/dashboard/ functions
+
+func CreateMentorDashboard(mentor models.Mentor, db *gorm.DB) MentorDashboard {
+	var projects []models.Project
+	var projectsInfo []ProjectInfo
+	var students []StudentInfo
+
+	db.Table("projects").
+		Where("mentor_id = ? OR secondary_mentor_id = ?", mentor.ID, mentor.ID).
+		Select("name", "repo_link", "commit_count", "pull_count", "lines_added", "lines_removed", "contributors", "pulls", "project_status").
+		Find(&projects)
+
+	for _, project := range projects {
+		pulls := make([]string, 0)
+		if len(project.Pulls) != 0 {
+			pulls = strings.Split(project.Pulls, ",")
+		}
+
+		projectInfo := ProjectInfo{
+			Name:          project.Name,
+			RepoLink:      project.RepoLink,
+			ProjectStatus: project.ProjectStatus,
+
+			CommitCount:  project.CommitCount,
+			PullCount:    project.PullCount,
+			LinesAdded:   project.LinesAdded,
+			LinesRemoved: project.LinesRemoved,
+
+			Pulls: pulls,
+		}
+		projectsInfo = append(projectsInfo, projectInfo)
+
+		var student StudentInfo
+		for _, studentUsername := range strings.Split(project.Contributors, ",") {
+			db.Table("students").Where("username = ?", studentUsername).
+				Select("name", "username").First(&student)
+
+			containsStud := false
```

Comment:
Use a hash map to optimize this (as discussed). If there are any duplicates, find them before fetching them from the database.